pkg/actors/targets/workflow/state.go

```diff
@@ -0,0 +1,187 @@
+	w.setOrchestrationMetadata(w.rstate, w.getExecutionStartedEvent(state))
+	w.ometaBroadcaster.Broadcast(w.ometa)
+	return nil
+}
+
+// This method cleans up a workflow associated with the given actorID
+func (w *workflow) cleanupWorkflowStateInternal(ctx context.Context, state *wfenginestate.State, requiredAndNotCompleted bool) error {
+	// If the workflow is required to complete but it's not yet completed then return [ErrNotCompleted]
+	// This check is used by purging workflow
+	if requiredAndNotCompleted {
+		return api.ErrNotCompleted
+	}
+
+	// This will create a request to purge everything
+	req, err := state.GetPurgeRequest(w.actorID)
+	if err != nil {
+		return err
+	}
+
+	// This will do the purging
+	err = w.actorState.TransactionalStateOperation(ctx, true, req)
+	if err != nil {
+		return err
+	}
+
+	w.table.DeleteFromTableIn(w, 0)
+	w.cleanup()
+
+	return nil
+}
+
+func (w *workflow) setOrchestrationMetadata(rstate *backend.OrchestrationRuntimeState, startEvent *protos.ExecutionStartedEvent) {
+	var se *protos.ExecutionStartedEvent = nil
+	if rstate.GetStartEvent() != nil {
+		se = rstate.GetStartEvent()
+	} else if startEvent != nil {
+		se = startEvent
+	}
+
+	name, _ := runtimestate.Name(rstate)
+	if name == "" && se != nil {
+		name = se.GetName()
+	}
+	createdAt, _ := runtimestate.CreatedTime(rstate)
+	lastUpdated, _ := runtimestate.LastUpdatedTime(rstate)
+	completedAt, _ := runtimestate.CompletedTime(rstate)
+	input, _ := runtimestate.Input(rstate)
+	output, _ := runtimestate.Output(rstate)
+	failureDetails, _ := runtimestate.FailureDetails(rstate)
+	var parentInstanceID string
+	if se != nil && se.GetParentInstance() != nil && se.GetParentInstance().GetOrchestrationInstance() != nil {
+		parentInstanceID = se.GetParentInstance().GetOrchestrationInstance().GetInstanceId()
+	}
+	w.ometa = &backend.OrchestrationMetadata{
+		InstanceId:       rstate.GetInstanceId(),
+		Name:             name,
+		RuntimeStatus:    runtimestate.RuntimeStatus(rstate),
+		CreatedAt:        timestamppb.New(createdAt),
+		LastUpdatedAt:    timestamppb.New(lastUpdated),
+		CompletedAt:      timestamppb.New(completedAt),
+		Input:            input,
+		Output:           output,
+		CustomStatus:     rstate.GetCustomStatus(),
+		FailureDetails:   failureDetails,
+		ParentInstanceId: parentInstanceID,
+	}
+}
+
+func (w *workflow) cleanup() {
+	w.lock.Lock()
+	defer w.lock.Unlock()
+
+	w.ometaBroadcaster.Close()
+	w.state = nil // A bit of extra caution, shouldn't be necessary
+	w.rstate = nil
+	w.ometa = nil
+
+	if w.closed.CompareAndSwap(false, true) {
+		close(w.closeCh)
+	}
+}
+
+// This method purges all the completed activity data from a workflow associated with the given actorID
+func (w *workflow) purgeWorkflowState(ctx context.Context) error {
+	state, _, err := w.loadInternalState(ctx)
+	if err != nil {
+		return err
+	}
+	if state == nil {
+		return api.ErrInstanceNotFound
```

Comment:
same comment here - is `ErrInstanceNotFound` correct to return here? Feel like its a bit misleading